### PR TITLE
Cjg/0.7 2

### PR DESCRIPTION
**Pull request by Cody-G**
This gets tests passing and depwarns removed for Julia 0.7.  I think I'll need to update the travis script too.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,10 +1,13 @@
-julia 0.6
-JSON 0.17.2 0.17.2+
-AxisArrays 0.2.1 0.2.1+
-IntervalSets 0.2.0 0.2.0+
-Unitful 0.8.0 0.8.0+
-MappedArrays 0.0.7 0.0.7+
-Ranges 0.0.1 0.0.1+
-DataStructures 0.8.4 0.8.4+
-DSP 0.5.1 0.5.1+
+julia 0.7
```
Comment (timholy) [inline REQUIRE]: You don't need both Project.toml and REQUIRE; you can just delete the REQUIRE

File: deps/build.jl [hunk 2]
```diff
@@ -1,9 +1,10 @@
-pkgs = ["UnitAliases", "ImagineHardware"]
-
-for p in pkgs
-    try
-        Pkg.installed(p)
-    catch
-        Pkg.clone("https://github.com/HolyLab/"*p*".jl.git")
-    end
-end
+#using Pkg
```
Comment (timholy) [inline deps/build.jl]: Can also be deleted

File: src/convenience.jl [hunk 3]
```diff
@@ -1,75 +1,85 @@
...
-
-isgalvomonitor{T<:ImagineSignal}(com::T)  = isgalvomonitor(daq_channel(com), rig_name(com))
-findgalvomonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = find(x->isgalvomonitor(x), coms)
-getgalvomonitors{T<:ImagineSignal}(coms::AbstractVector{T}) = view(coms, findgalvomonitors(coms))
-
-hasmonitor{T<:ImagineSignal}(com::T) = iscam(com) || ispos(com) || isgalvo(com)
-hasactuator{T<:ImagineSignal}(com::T) = iscammonitor(com) || isposmonitor(com) || isgalvomonitor(com)
-
-function monitor_name{T<:ImagineSignal}(com::T)
+finddaqchan(coms::AbstractVector{T}, nm::AbstractString) where{T<:ImagineSignal} =
```
Comment (timholy) [inline src/convenience.jl]: I usually see a space after `where` but clearly it must not be essential. It's pretty readable this way.

File: src/convenience.jl [hunk 4]
```diff
@@ -1,75 +1,85 @@
...
+isanalog(com::T) where{T<:ImagineSignal} =
+    isanalog(daq_channel(com), rig_name(com))
+findanalog(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    findall(x->!isdigital(x), coms)
+getanalog(coms::AbstractVector{T}) where{T<:ImagineSignal} =
+    view(coms, findanalog(coms))
+
+isoutput(com::T) where{T<:ImagineSignal} = isoutput(daq_channel(com), rig_name(com))
+isinput(com::T) where{T<:ImagineSignal} = !isoutput(com)
+findoutputs(coms::AbstractVector{T}) where{T<:ImagineSignal} = findall(x->isoutput(x), coms)
```
Comment (timholy) [inline src/convenience.jl]: `findall(x->isoutput(x), coms)` can be more simply written as `findall(isoutput, coms)`. You're less likely to get weird `#3##4(x)` entries in your backtraces, too :smile:

**Comment by Cody-G** [comment]
Thanks Tim for the comments!  I addressed them but I still can't seem to get Travis to work correctly with the private registry.  (see https://github.com/HolyLab/HolyLabRegistry/issues/26)

**Comment by Cody-G** [comment]
Note the bad coverage result in the bot's comment above is due to https://github.com/JuliaCI/Coverage.jl/issues/187.  I worked around this by using an old version of Coverage.jl for now.  Crossref #69.

**Pull request merged**